app/lib/atomic_lti/services/names_and_roles.rb

```diff
@@ -61,6 +61,19 @@ def list(query: {}, page_url: nil)
         )
       end
 
+      def list_all(query: {})
+        page_body = nil
+
+        members = AtomicLti::PagingHelper.paginate_request do |next_link|
+          result_page = list(query: query, page_url: next_link)
+          page_body = JSON.parse(result_page.body)
```

Comment:
This is probably feature creep because I think this was a problem before. It may be worth putting a try catch around the JSON.parse so if  canvas doesn't send a body in an error case it doesn't blow up. I think that has been a problem in the past.